avalon/pipeline.py

```diff
@@ -248,6 +248,56 @@ def process(self, session, **kwargs):
         pass
 
 
+class InventoryAction(object):
+    """A custom action for the scene inventory tool
+
+    If registered the action will be visible in the Right Mouse Button menu
+    under the submenu "Actions".
+
+    """
+
+    label = None
+    icon = None
+    color = None
+    order = 0
+
+    @staticmethod
+    def is_compatible(container):
+        """Override function in a custom class
+
+        This method is specifically used to ensure the action can operate on
+        the container.
+
+        Args:
+            container: dict
+
+        Returns:
+            bool
+        """
+        return True
+
+    def process(self, containers):
+        """Override function in a custom class
+
+        This method will receive all containers even those which are
+        incompatible. It is advised to create a small filter along the lines
+        of this example:
+
+        valid_containers = [c for c in containers if "myKey" in c]
+
+        The return value will need to be a True-ish value to trigger
+        the data_changed signal in order to refresh the view.
+
+        Args:
+            containers: collection if dictionaries
```

Comment:
containers (list): list of container dictionaries

Reply:
We call them containers but they are dictionaries. If I read the docs this way I understand that it is expected of me to create a list of dictionaries which are referred to as container in Avalon jargon but that does not change their type.

Reply:
It needs to be a list. It's as simple as that. The `(list)` is a hint for the IDE to know the type it needs, plus it his how the Google Docstrings work. :) 

Unfortunately my correction is the correct way of doing it, but you're free to put whatever after the `container (list):` - but for that I basically used your version, refactored collection to list because collection doesn't mean much in Python-speak and fixed the typo `if` to `of`.

Reply:
I thought you were mainly point out the descriptor of the argument and not the name and type of the argument